### PR TITLE
ERC-1504 Upgradable Smart Contract (USC)

### PR DESCRIPTION
**Pull request by swordghost**
PR for Upgradable Smart Contract.

Discussions at https://github.com/ethereum/EIPs/issues/1503

### CODE REVIEW

**Comment by Arachnid** [comment]
Does this need to be a standard? It seems like it would do just fine as a well documented library people can reuse.

File: EIPS/eip-1504.md [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+
+## Specification
+
+The upgradable contract consists of three parts:
+
+- **Handler contract** (implements **Handler interface**) defines operations and provides services. This contract can be upgraded;
+- **Data contract** keeps the resources (data) and is controlled by the Handler contract;
+- **Upgrader contract (optional)** deals with the voting mechanism and upgrades the Handler contract. The voters are pre-defined by the contract owner. 
+
+> The following codes are exact copies of the [ERC-54 Upgradable Smart Contract.](https://gist.github.com/swordghost/969648793ed54fdf027cb6df1b894b75)
```
Comment (axic) [inline EIPS/eip-1504.md]: There is no such thing as "ERC-54", at least I cannot find such a number in this repo.

Comment (swordghost) [reply EIPS/eip-1504.md]: The mistake has been corrected.

**Comment by Arachnid** [comment]
Please fix the outstanding build errors (spelling check).

**Comment by swordghost** [comment]
> Please fix the outstanding build errors (spelling check).

I'm sorry, the bug is fixed.

**Comment by axic** [comment]
This PR still has ERC-54 in at least two more places.

**Comment by swordghost** [comment]
> This PR still has ERC-54 in at least two more places.

Sorry, we'll fix it soon. Thanks for your help.

**Pull request merged**